lib/pbench/test/unit/common/test_logging.py

```diff
@@ -0,0 +1,144 @@
+        ) as TMP:
+            assert Path(TMP).exists()
+            self.config.log_dir = str(Path(TMP) / "log-dir")
+            assert not Path(
+                self.config.log_dir
+            ).exists(), (
+                f"Test logging directory unexpectedly exists, {self.config.log_dir}"
+            )
+            Path(self.config.log_dir).mkdir()
+            self.logger = get_pbench_logger(fname, self.config)
+            assert Path(
+                self.config.log_dir
+            ).is_dir(), f"Missing logging directory, {self.config.log_dir}"
+            assert _handlers[fname] == self.logger.logger.handlers[0]
+            assert (
+                _handlers[fname].__class__.__name__ == "FileHandler"
+            ), f"Unexpected handler set, {_handlers[fname].__class__.__name__!r}"
+
+    def test_log_messages_to_devlog(self):
+        """Test to log messages via /dev/log"""
+        # Was test-26.2, via test_logger_type.py
+        fname = "test_log_messages_to_devlog"
+        self.logger = get_pbench_logger(fname, self.config)
+        assert (
+            self.config.logger_type == "devlog"
+        ), f"Unexpected logger type encountered, '{self.config.logger_type}', expected 'devlog'"
+        assert (
+            self.config.log_dir is None
+        ), f"Unexpected log directory configuration found, {self.config_log_dir}"
+        assert _handlers[fname] == self.logger.logger.handlers[0]
+        assert (
+            _handlers[fname].__class__.__name__ == "SysLogHandler"
+        ), f"Unexpected handler set for {fname}, {_handlers[fname].__class__.__name__!r}"
+        assert (
+            _handlers[fname].address == "/dev/log"
+        ), f"Unexpected handler address set, {_handlers[fname].address!r}"
+
+    def test_log_messages_to_hostport(self):
+        """Test to check error when logger_port and logger_host are not provided with "hostport"."""
+        # Was test-26.3, via test_logger_type.py
+        fname = "test_log_messages_to_hostport"
+        self.config.logger_type = "hostport"
+        self.config.logger_host = "host.example.com"
+        self.config.logger_port = "42"
+        self.logger = get_pbench_logger(fname, self.config)
+        assert (
+            self.config.logger_type == "hostport"
+        ), f"Unexpected logger type encountered, '{self.config.logger_type}', expected 'hostport'"
+        assert (
+            self.config.log_dir is None
+        ), f"Unexpected log directory configuration found, {self.config_log_dir}"
+        assert _handlers[fname] == self.logger.logger.handlers[0]
+        assert (
+            _handlers[fname].__class__.__name__ == "SysLogHandler"
+        ), f"Unexpected handler set, {_handlers[fname].__class__.__name__!r}"
+        assert (
+            _handlers[fname].address[0] == "host.example.com"
+            and _handlers[fname].address[1] == 42
+        ), f"Unexpected handler address set, {_handlers[fname].address!r}"
+
+    @staticmethod
```

Comment:
Is there a non-obvious reason why this test case is static while the others are not?

Reply:
I'll be changing this to a non-static method, but the reason it is different is that all the other tests rely on the `config` fixture at line 17 to setup `self.config`.  I'll fix the code to stop using `self.logger` since it is not necessary.  This method creates its own config.